src/node.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Entry point for setting up of Requests Debugger Tool.
+ * Initiates actions such as processing of args, setting up loggers,
+ * initiating all connectivity checks and stats collection before starting
+ * the proxy tool.
+ */
+
+var constants = require('../config/constants');
+var LogFiles = constants.LOGS;
+var RdGlobalConfig = constants.RdGlobalConfig;
+var CommandLineManager = require('./commandLine');
+var ConnectivityChecker = require('./connectivity');
+var RdHandler = require('./server');
+var StatsFactory = require('./stats/statsFactory');
+var LogManager = require('./logger');
+var fs = require('fs');
+var path = require('path');
+var Utils = require('./utils');
+
+var RdTool = {
+
+  /**
+   * Initializes the Logging directory, Loggers & Stats Handlers.
+   */
+  initLoggers: function () {
+    var basePath = RdGlobalConfig.logsPath ? path.resolve(RdGlobalConfig.logsPath) : process.cwd();
+    RdGlobalConfig.LOGS_DIRECTORY = path.resolve(basePath, 'RequestsDebuggerLogs');
+
+    if (RdGlobalConfig.deleteExistingLogs) {
+      var filesToDelete = Object.keys(LogFiles).map(function (key) { return LogFiles[key]; });
+      filesToDelete.forEach(function (file) {
+        try {
+          fs.unlinkSync(path.resolve(RdGlobalConfig.LOGS_DIRECTORY, file));
+          /* eslint-disable-next-line no-empty */
+        } catch (e) {}
+      });
+    }
+
+    try {
+      fs.mkdirSync(RdGlobalConfig.LOGS_DIRECTORY);
+    } catch (e) {
+      if (e.code !== 'EEXIST') {
+        var errorMessage = "Error in creating 'RequestsDebuggerLogs' folder at path: " + basePath + "\n"
+                           + "Message: " + e.toString() + "\n";
+        console.log(errorMessage);
+        process.exit(1);
+      }
+    }
+
+    RdGlobalConfig.NetworkLogger = LogManager.initializeLogger(path.resolve(RdGlobalConfig.LOGS_DIRECTORY, LogFiles.NETWORK));
+    RdGlobalConfig.MemLogger = LogManager.initializeLogger(path.resolve(RdGlobalConfig.LOGS_DIRECTORY, LogFiles.MEM));
+    RdGlobalConfig.CPULogger = LogManager.initializeLogger(path.resolve(RdGlobalConfig.LOGS_DIRECTORY, LogFiles.CPU));
+    RdGlobalConfig.ReqLogger = LogManager.initializeLogger(path.resolve(RdGlobalConfig.LOGS_DIRECTORY, LogFiles.REQUESTS));
+    RdGlobalConfig.ConnLogger = LogManager.initializeLogger(path.resolve(RdGlobalConfig.LOGS_DIRECTORY, LogFiles.CONNECTIVITY));
+    RdGlobalConfig.ErrLogger = LogManager.initializeLogger(path.resolve(RdGlobalConfig.LOGS_DIRECTORY, LogFiles.ERROR));
+
+    RdGlobalConfig.NetworkLogHandler = function (topic, uuid, callback) {
+      topic = topic || 'NO_TOPIC';
```

Comment:
Moving it to constants. `DEFAULT_TOPIC` usage came seems different. `NO_TOPIC` is logged since it expected a topic but didn't get any. There need not be a `default` case in this. It would be great if topic is passed always.

Reply:
moved.

Reply:
Resolving.